app/Controllers/Event.php

```diff
@@ -22,6 +24,27 @@ public function get(int|null $year = null) {
         $speakerModel = new SpeakerModel();
         $speakers = $speakerModel->getPublished($event['id']);
 
+        $socialMediaTypeModel = new SocialMediaTypeModel();
+        $socialMediaTypes = $socialMediaTypeModel->all();
+
+        $socialMediaLinkModel = new SocialMediaLinkModel();
+        foreach ($speakers as &$speaker) {
+            $dbLinks = $socialMediaLinkModel->get_by_speaker_id($speaker['id']);
+            $links = [];
+            foreach ($dbLinks as $dbLink) {
+                $typeId = $dbLink['social_media_type_id'];
+                if (!isset($socialMediaTypes[$typeId])) {
+                    return $this->response->setStatusCode(500);
+                }
+                $type = $socialMediaTypes[$typeId];
+                $links[] = [
+                    'type' => $type['name'],
+                    'url' => $dbLink['url'],
+                ];
+            }
+            $speaker['social_media_links'] = $links;
+        }
```

Comment:
This will have a `n` amount of queries to the db. It could be optimized to query all social media links for all speakers in `$speakers` at once.
I expect there to be no "big" problems with doing it like it currently is, as there are not that many speakers per event.

Reply:
Yes, you're right. Although I don't think this would be a performance problem, I will change that.